platforms/jvm/language-java/src/integTest/groovy/org/gradle/java/compile/incremental/CrossTaskIncrementalJavaCompilationIntegrationTest.groovy

```diff
@@ -55,9 +55,11 @@ abstract class CrossTaskIncrementalJavaCompilationIntegrationTest extends Abstra
         """
         file("impl/build.gradle") << """
             def layout = project.layout
-            compileJava.doFirst {
+            compileJava {
                 options.compilerArgs << "--module-path" << classpath.join(File.pathSeparator)
-                classpath = layout.files()
+                doFirst {
+                    classpath = layout.files()
+                }
```

Comment:
❓ Why is this necessary/better now?

Reply:
Ah, right, but in the other PR we fixed this differently. Let's merge that one, and then this will not be needed, right?

Reply:
Yes, we can handle it in the same way